hrms/hr/doctype/full_and_final_statement/full_and_final_statement.py

```diff
@@ -310,3 +313,9 @@ def update_full_and_final_statement_status(doc, method=None):
 			fnf = frappe.get_doc("Full and Final Statement", entry.reference_name)
 			fnf.db_set("status", status)
 			fnf.notify_update()
+
+
+def update_status_of_reference_documents(doc, status="Paid"):
```

Comment:
We can encapsulate this function I think. No need to keep it outside the class
Gratuity holds a "paid_amount" field. We should update that and call `gratuity.set_status` to update the status. If we just update the status, `paid_amount` field will remain 0

Reply:
Also can you add a test 🙈